testservices/neutronservice/service_http.go

```diff
@@ -0,0 +1,719 @@
+// Neutron double testing service - HTTP API implementation
+
+package neutronservice
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strconv"
+	"strings"
+
+	"gopkg.in/goose.v1/neutron"
+	"gopkg.in/goose.v1/testservices"
+	"gopkg.in/goose.v1/testservices/identityservice"
+)
+
+const (
+	authToken               = "X-Auth-Token"
+	apiFloatingIPsV2        = "/v2.0/" + neutron.ApiFloatingIPsV2
+	apiNetworksV2           = "/v2.0/" + neutron.ApiNetworksV2
+	apiSubnetsV2            = "/v2.0/" + neutron.ApiSubnetsV2
+	apiSecurityGroupsV2     = "/v2.0/" + neutron.ApiSecurityGroupsV2
+	apiSecurityGroupRulesV2 = "/v2.0/" + neutron.ApiSecurityGroupRulesV2
+)
+
+// errorResponse defines a single HTTP error response.
+type errorResponse struct {
+	code        int
+	body        string
+	contentType string
+	errorText   string
+	headers     map[string]string
+	neutron     *Neutron
+}
+
+// verbatim real Neutron responses (as errors).
+var (
+	errUnauthorized = &errorResponse{
+		http.StatusUnauthorized,
+		`401 Unauthorized
+
+This server could not verify that you are authorized to access the ` +
+			`document you requested. Either you supplied the wrong ` +
+			`credentials (e.g., bad password), or your browser does ` +
+			`not understand how to supply the credentials required.
+
+ Authentication required
+`,
+		"text/plain; charset=UTF-8",
+		"unauthorized request",
+		nil,
+		nil,
+	}
+	errForbidden = &errorResponse{
+		http.StatusForbidden,
+		`{"forbidden": {"message": "Policy doesn't allow compute_extension:` +
+			`flavormanage to be performed.", "code": 403}}`,
+		"application/json; charset=UTF-8",
+		"forbidden flavors request",
+		nil,
+		nil,
+	}
+	errBadRequest = &errorResponse{
+		http.StatusBadRequest,
+		`{"badRequest": {"message": "Malformed request url", "code": 400}}`,
+		"application/json; charset=UTF-8",
+		"bad request base path or URL",
+		nil,
+		nil,
+	}
+	errBadRequest2 = &errorResponse{
```

Comment:
I'd try and give these clearer names, eg errBadRequestMalformedURL/errBadRequestIncorrect/errBadRequestMalformedBody.